### ISSUE TITLE
Test/Inference code

### ISSUE THREAD

**Issue by TopTea1** [JacobYuan7/RLIP#2]
Hi @JacobYuan7, first of all, thanks for sharing your work!

I had a question about generating HOIs on new images without annotations, is it possible to modify generate_vcoco_official.py to do this?

**Comment by JacobYuan7** [comment]
@TopTea1 
Thanks for your interest.
You are right. I am also implementing one version of _inference_on_custom_imgs_hico.py_ to allow for inferring on custom images placed in a folder 'custom_imgs/' and saving detection results. I will try to make it available online very soon.

**Comment by TopTea1** [comment]
Thank you @JacobYuan7 !

**Comment by JacobYuan7** [comment]
@TopTea1 
I updated the link of the pre-trained model in Inference on Custom Images. This updated one is fine-tuned on HICO. The old one is the pre-trained parameters. Sorry about the mistake.

**Comment by TopTea1** [comment]
Thanks for the update @JacobYuan7, after using the code, a question comes to me, should ```verb_scores``` be considered as confidence for this type of interaction ? Or how should it be considered?
Thanks !

**Comment by JacobYuan7** [comment]
@TopTea1 
Yes. The verb_scores are the multiplication of original verb_scores and object scores. Thus, it can be interpreted as the confidence for this relation triplet.

The following information is for your reference:
the saved detection results are post-processed by `results = postprocessors['hoi'](outputs, orig_target_sizes)`. Thus, you can refer to class `PostProcessHOI(nn.Module)` in Line 3701 of models/hoi.py. Five values are stored in the dictionary: labels (subject and object labels), boxes (subject and object boxes), sub_ids, obj_ids and verb_scores. The verb_scores are generated by multiplying original verb_scores by object scores using `vs = vs * os.unsqueeze(1)`.

**Comment by TopTea1** [comment]
Thanks @JacobYuan7 for your answer, when I looked at the value of `vs` is always very low (for example 1e-4) is this normal? Could you give an example of a usable threshold? Thanks!

**Comment by JacobYuan7** [comment]
@TopTea1 
In most cases, many values can be low because the number of query pairs is 64 on HICO-DET, which is usually greater than the number of relation triplets in one image.

Since the verb_scores are generated by multiplying original verb_scores by object scores, thus we can start from `threshold = 0.5*0.5 = 0.25`. To have a better performance on your task, you can traverse the threshold value.

**Comment by JacobYuan7** [comment]
@TopTea1 
I closed this issue as completed. Feel free to re-open it if you have further questions.
